### PR TITLE
Switch beam model by map label field

### PR DESCRIPTION
**Pull request by at-wat**

### CODE REVIEW

**Comment by at-wat** [comment]
## [[#839](https://travis-ci.org/at-wat/mcl_3dl/builds/722473698)] PASSED on noetic

Tested on Alpine ROS

**Comment by at-wat** [comment]
## [[#840](https://travis-ci.org/at-wat/mcl_3dl/builds/722474655)] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_cloud_accum.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_imu_measurement_model_gravity.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_motion_prediction_model_differential_drive.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_noise_generator.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 12 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_state_6dof.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 10 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-localization_rostest__without_imu_true.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_debug_output_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_landmark_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 2 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_debug_output.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_landmark.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 8 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 2 tests
Summary: 114 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
Labeling semi-transparent wall to tell mcl_3dl not to use them in beam-model ray casting.

With label | Without label
--- | ---
mcl-semitrans2 | mcl-semitrans1

**Comment by at-wat** [comment]
## [[#841](https://travis-ci.org/at-wat/mcl_3dl/builds/722645585)] FAILED on noetic

Tested on Alpine ROS

**Comment by at-wat** [comment]
@nhatao please take a look

File: src/lidar_measurement_model_beam.cpp [hunk 1]
```diff
@@ -87,6 +87,10 @@ void LidarMeasurementModelBeam::loadConfig(
   double ang_total_ref;
   pnh.param("ang_total_ref", ang_total_ref, M_PI / 6.0);
   sin_total_ref_ = sinf(ang_total_ref);
+
+  int use_label_leq;
+  pnh.param("use_label_leq", use_label_leq, static_cast<int>(0xFFFFFFFF));
```
Comment (nhatao) [inline src/lidar_measurement_model_beam.cpp:92]: is this parameter name correct? It sounds like a boolean value, but actually it is used as a threshold.

Comment (at-wat) [reply src/lidar_measurement_model_beam.cpp:92]: `filter_label_if_leq` or `filter_label_max`? What do you think?

Comment (nhatao) [reply src/lidar_measurement_model_beam.cpp:92]: `filter_label_max` or `filter_label_leq`?

**Comment by at-wat** [comment]
## [[#842](https://travis-ci.org/at-wat/mcl_3dl/builds/722987079)] PASSED on noetic

Tested on Alpine ROS

**Comment by at-wat** [comment]
## [[#843](https://travis-ci.org/at-wat/mcl_3dl/builds/722994744)] PASSED on noetic

Tested on Alpine ROS

**Review by nhatao — APPROVED** [review]

**Pull request merged**